File: hrmanagementfrontend/src/app/type/Stagiaire.ts
import { TacheAssignee } from './TacheAssignee';
export interface Stagiaire{
    id?:number;
    nom:string;
    prenom:string;
    age:number;
    adresse:string;
    departement:string;
    dateDebut:Date;
    dateFin:Date;
    tachesAssignees:TacheAssignee[];
}

File: hr-management/src/main/java/com/project/hrmanagement/entity/Stagiaire.java
package com.project.hrmanagement.entity;

import java.time.LocalDate;
import java.util.List;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "t_stagiaire")
public class Stagiaire {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nom;
    private String prenom;
    private Long age;
    private String adresse;
    private String departement;
    private LocalDate dateDebut;
    private LocalDate dateFin;
    @OneToMany(fetch = FetchType.EAGER)
    private List<TacheAssignee> tachesAssignees;
}